src/app/footer/footer.component.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { FooterComponent } from './footer.component';
+import {ContactService} from '../services/contact.service';
+import {RouterLinkStubDirective} from '../../testing/router-stubs';
+import {Component, DebugElement} from '@angular/core';
+import {By} from '@angular/platform-browser';
+
+
+describe('FooterComponent', () => {
+  let component: FooterComponent;
+  let fixture: ComponentFixture<FooterComponent>;
+  let linkDes: DebugElement[];
+  let links: any;
+
+  const contactServiceStub = {
+    email: 'ContactMe'
+  };
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
```

Comment:
Testing the rendered UI are good (don't delete any tests) but I think there is a lot of value (maybe more) in testing the actual component class methods. In this case there are two unit tests for `copyRightYears` that you could do. Your rendered UI only covers 1 case.